### ISSUE TITLE
ProtectedOutputException is thrown for unprotected file

### ISSUE THREAD

**Issue by kpj** [snakemake/snakemake#10]
**Snakemake version**
5.7.0

**Describe the bug**
When trying to re-execute  a rule using `snakemake -R <rulename>`, snakemake throws an `ProtectedOutputException` even though the output file is not protected.
Downgrading to snakemake 5.6.0 solves the problem.

**Minimal example**
Consider the following Snakefile:
```
rule all:
    input:
        'out.txt'

rule dummy:
    output:
        'out.txt'
    shell:
        """
        touch {output}
        """
```

Executing it for the first time works as expected:
```bash
$ snakemake
Building DAG of jobs...
Using shell: /usr/local/bin/bash
Provided cores: 1
Rules claiming more threads will be scaled down.
Job counts:
	count	jobs
	1	all
	1	dummy
	2

[Tue Oct  8 13:57:07 2019]
rule dummy:
    output: out.txt
    jobid: 1

[Tue Oct  8 13:57:07 2019]
Finished job 1.
1 of 2 steps (50%) done

[Tue Oct  8 13:57:07 2019]
localrule all:
    input: out.txt
    jobid: 0

[Tue Oct  8 13:57:07 2019]
Finished job 0.
2 of 2 steps (100%) done
```

Re-executing `dummy` then fails:
```bash
$ snakemake -R dummy
Building DAG of jobs...
Using shell: /usr/local/bin/bash
Provided cores: 1
Rules claiming more threads will be scaled down.
Job counts:
	count	jobs
	1	all
	1	dummy
	2
ProtectedOutputException in line 5 of /private/tmp/example/Snakefile:
Write-protected output files for rule dummy:
out.txt
```

**Comment by openpaul** [comment]
I just tried this with snakemake 5.7.0 and the current master and could not reproduce this issue. I am on debian. Maybe this is some operating specific issue?

**Comment by smatsumt** [comment]
My environment got same issue in v5.7.0 (MacOS HighSierra version 10.13.6)

I guess the problem is caused by following code.

https://github.com/snakemake/snakemake/blob/44206be0bb278dca61a103e3b1fc873af976a554/snakemake/io.py#L282-L289

**Comment by smatsumt** [comment]
This issue is happened because `os.access` returns `False` even if the file exists with writable permission when `follow_symlinks=False`. I'm not sure it is only in MacOS, or same in any other environment.

```
$ cd snakemake
$ python
Python 3.7.4 (default, Jul  9 2019, 18:15:00)
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.access("README.md", os.W_OK, follow_symlinks=True)
True
>>> os.access("README.md", os.W_OK, follow_symlinks=False)
False
```

README.md exists, so the return value should be True...

**Comment by smatsumt** [comment]
Python on official docker image does not have above issue. 

```
$ docker run -it python /bin/sh
# echo foo > foo.txt
# python
Python 3.7.4 (default, Sep 12 2019, 15:40:15)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.access("foo.txt", os.W_OK, follow_symlinks=False)
True
```

The problem is only in MacOS, maybe.

**Comment by smatsumt** [comment]
I've wrote PR for fix it. I'm glad if I can hear your opinion.

**Comment by kpj** [comment]
I am using macOS Mojave 10.14.6.

After installing `snakemake-5.7.0+8.g59846dd7` the issue does no longer persist, thanks!

**Comment by johanneskoester** [comment]
Fixed by PR #29.
